doc/high_availability.md

```diff
@@ -1,110 +1,74 @@
-EnableHA API call
-
- Its worker loop continally watches 
-
- 1. The current set of controllers 
- 2. The addresses of the current controllers 
- 3. The status of the current mongo peergroup
- 
-It feeds all that information into `desiredPeerGroup`, which provides the peer 
-group that we want to be and continually tries to set that peer group in mongo 
-until it succeeds.
- 
-**NOTE:** There is one situation which currently doesn't work which is 
-that if you've only got one controller, you can't switch to another one.
-
-### The Singleton Workers
-
-**Note:** This section reflects the current behavior of these workers but 
-should by no means be taken as an example to follow since most (if not all)
-should run concurrently and are going to change in the near future.
-
-The following workers require only a single instance to be running
-at any one moment:
-
- * The environment provisioner
- * The firewaller
- * The charm revision updater
- * The state cleaner
- * The transaction resumer
- * The minunits worker
-
-When a machine agent connects to the state, it decides whether
-it is on the same instance as the mongo master instance, and
-if so, it runs the singleton workers; otherwise it doesn't run them.
-
-Because we are using `mgo.Strong` consistency semantics,
-it's guaranteed that our mongo connection will be dropped
-when the master changes, which means that when the
-master changes, the machine agent will reconnect to the
-state and choose whether to run the singleton workers again.
-
-It also means that we can never accidentally have two
-singleton workers performing operations at the same time.
+HA mode is invoked via the `enable-ha` command. By default, this will ensure 
+three controllers, but using the `-n` flag will allow running with five or 
+seven controllers.
+
+The number of controllers can be reduced by invoking `remove-machine` with a 
+controller machine ID, and increased by re-running the `enable-ha` command.
+
+### Dqlite
+
+Each controller is a Dqlite node. The `dbaccessor` worker on each controller is 
+responsible for maintaining the Dqlite cluster. When entering HA mode, the 
+`dbaccessor` worker will configure the local Dqlite node as a member of the 
+cluster.
+
+When starting Dqlite, the worker must bind it to an IP address on the local
+machine. It is a requirement that there is a unique local-cloud-scoped address
+for Dqlite to use. If there is no unique address, new nodes will not be joined 
+to the cluster until one can be determined. See _Controller Charm_ below.
+
+Beyond joining nodes to the cluster, Juju does not manage Dqlite node roles.
+This is handled within Dqlite itself. A cluster will have one leader, voters
+which are eligible to participate in leader elections, stand-bys and spares.
+
+Juju does not predicate any logic on node roles.
+
+If the number of controller instances is reduced to one, the `dbaccessor` 
+worker will recognise this scenario and proactively reconfigure the cluster to
```

Comment:
**suggestion:** I think `recognise` should be in the present tense.
```suggestion
worker recognises this scenario and proactively reconfigure the cluster to
```

References: https://www.enago.com/academy/ifthen-using-conditional-sentences-in-academic-writing/
(we are in a case where the **Zero conditional sentences** case => there is no assumption on the behavior of `dbaccessor` in the case of the number of controller instances drop to one. It is a known, sure consequences (at least if we have no bug :) )

**suggestion:** maybe search `will` usages and assess it regarding above rules ;)